addons/website/models/digest.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class Digest(models.Model):
+    _inherit = 'digest.digest'
+
+    kpi_website_visitor_cnt = fields.Boolean('Visitors')
+    kpi_website_visitor_cnt_value = fields.Integer(compute='_compute_kpi_website_visitors_cnt_value')
+    kpi_tracked_page_view_cnt = fields.Boolean('Tracked page views')  # Non uniq
+    kpi_tracked_page_view_cnt_value = fields.Integer(compute='_compute_kpi_tracked_page_view_cnt_value')
+
+    def _compute_visitor_cnt(self, ctx):
+        website_ids, start, end = ctx
+        if len(website_ids) == 0:
+            return 0
+        # The query takes into account that some website_visitor are the same visitor through parent_id
+        self._cr.execute("""
+            SELECT count(distinct COALESCE(v.parent_id, v.id))
+                FROM website_visitor AS v
+                JOIN website_track AS t ON t.visitor_id = v.id AND t.visit_datetime >= %(start)s AND t.visit_datetime < %(end)s
```

Comment:
Can't we use last_connection_datetime and skip the join on track pages (aka could even maybe be a standard domain, with active_test flag in search) ?

Reply:
With only the last_connection_datetime, we don't have the connection history and the kpi system run multiple queries with different time interval to compute the percentage of increase or decrease. It could lead to strange results (visitor going from an time interval to another).